include/hipSYCL/sycl/queue.hpp

```diff
@@ -741,6 +761,71 @@ class queue : public detail::property_carrying_object
     });
   }
 
+  
+  // Explicit copy functions
+  template <typename T, int dim, access::mode mode, access::target tgt,
+            accessor_variant variant>
+  event copy(accessor<T, dim, mode, tgt, variant> src,
+             shared_ptr_class<T> dest) {
+    return this->submit([&](sycl::handler &cgh) {
+      cgh.copy(src, dest);
+    });           
+  }
+  
+  template <typename T, int dim, access::mode mode, access::target tgt,
+            accessor_variant variant>
+  event copy(shared_ptr_class<T> src,
+             accessor<T, dim, mode, tgt, variant> dest) {
+    return this->submit([&](sycl::handler &cgh) {
+      cgh.copy(src, dest);
+    });           
+  }
+
+  template <typename T, int dim, access::mode mode, access::target tgt,
+            accessor_variant variant>
+  event copy(accessor<T, dim, mode, tgt, variant> src,
+             T *dest) {
+    return this->submit([&](sycl::handler &cgh) {
+      cgh.copy(src, dest);
+    });     
+  }
+
+  template <typename T, int dim, access::mode mode, access::target tgt,
+            accessor_variant variant>
+  event copy(const T *src,
+             accessor<T, dim, mode, tgt, variant> dest) {
+    return this->submit([&](sycl::handler &cgh) {
+      cgh.copy(src, dest);
+    });             
+  }
+
+  template <typename T, int dim, access::mode srcMode, access::mode dstMode,
+            access::target srcTgt, access::target destTgt,
+            accessor_variant VariantSrc, accessor_variant VariantDest>
+  event copy(accessor<T, dim, srcMode, srcTgt, VariantSrc> src,
+             accessor<T, dim, dstMode, destTgt, VariantDest> dest) {
+    return this->submit([&](sycl::handler &cgh) {
+      cgh.copy(src, dest);
+    });  
+  }
+
+  template <typename T, int dim, access::mode mode, access::target tgt,
+            accessor_variant variant>
+  event update_host(accessor<T, dim, mode, tgt, variant> acc) {
```

Comment:
If we expose `update_host`, we should probably also expose hipSYCL's device `update()` extension (`handler::update()`)

Reply:
Done.